[Jack] Ladder step3

somedd opened the pull request.

somedd committed:
Generate structure for dividing functions

1. LadderGame, LadderPlayer, InputView, ResultView기능을 위해 구조체로
나누었습니다.

somedd committed:
Add init-function of LadderGame

somedd committed:
Reduce depth of main

somedd committed:
Add function for printing player's names

godrm commented:
구조체 역할 분담은 전체적으로 잘 나누었습니다. 
객체를 기능별 역할과 책임으로 나누고 나면, 객체 사이에 전달하는 데이터구조를 고민해보세요.
지금 사다리 정보를 표현하는 2차원 배열을 [[String]] 타입으로 전달하고 있는데,
문자열보다는 원래 사다리 유무 정보를 표현하는 [[Bool]] 타입이 더 효과적입니다. 
사다리 정보를 받아서 원하는 문자로 출력하는 것은 OutputView 혹은 ResultView 의 역할입니다.
객체 사이에 전달하는 데이터 구조를 개선해보세요.

somedd committed:
Change data structure of ladder

1. 사다리구조를 2차원문자열배열에서 2차원 불타입배열로 바꾸었습니다.
2. outputView구조체에서는 랜덤 사다리를 만들고, ResultView구조체에서
해당 사다리를 문자열로 바꾸어 출력하도록 수정하였습니다.

godrm commented on LadderGame/LadderGame/LadderGame.swift:
```diff
@@ -23,62 +23,45 @@ struct LadderGame {
+        guard makeVal == false || RandomNum == 0 else {
+        return true
+        }
+        return false
```
이 정도 비교구문은 guard else 를 쓰기보다는 if - else 를 써도 되고, 
더 단순하게는 `return !(makeVal == false || RandomNum == 0)` 형태도 가능할 것 같습니다. 
if 문과 guard 문을 선택하는 기준을 정해보세요.

godrm commented on LadderGame/LadderGame/LadderGame.swift:
```diff
@@ -23,62 +23,45 @@ struct LadderGame {
+        for index in 0..<oneFloorOfLadder.count {
+            guard index == 0 || temp[index - 1] != true else {
+                temp[index] = makeRandomHorizontalLine(false)
                 continue
```
여기 guard else 구문도 if 구문이 나아보입니다. 앞서 작성한 피드백과 함께 비교해보세요.

godrm commented on LadderGame/LadderGame/ResultView.swift:
```diff
@@ -9,17 +9,44 @@
+                temp[index] = empty
+                continue
+            }
+            temp[index] = bar
```
여기도 for-in 내부에서는 guard else 구문 보다는 if-else 구문이 흐름이 더 명확할 것 같습니다.

somedd committed:
Edit sources of struct LadderGame and ResultView

1. guard문을 if문으로 바꾸어 보았습니다.
2. 일부 변수명이 명확하지 않은 변수들을 수정했습니다.

godrm commented on LadderGame/LadderGame/LadderGame.swift:
```diff
@@ -31,10 +31,7 @@ struct LadderGame {
-        return true
-        }
-        return false
+        return !(makeVal == false || RandomNum == 0)
```
조건식을 쓸 때는 읽으면서 조건을 판단하기 쉬운게 좋습니다. 
`!(makeVal == false || RandomNum == 0)` 과 동치인 `(makeVal != false && RandomNum != 0)` 형태도 늘 같이 고민해보세요. 참고하세요.

somedd replied:
넵!

godrm commented:
전체적으로 잘 정리하셨습니다. 다음 단계는 단위 테스트 코드를 경험하는 단계입니다.
이번 단계에서 만든 struct별로 각 기능 단위로 테스트하는 코드를 작성해보세요.
